src/Orleans/Configuration/ClientConfiguration.cs

```diff
@@ -74,6 +74,13 @@ public enum GatewayProviderType
         public string DataConnectionString { get; set; }
 
         /// <summary>
+        /// the id of the cluster to which this client is connecting. 
+        /// This argument is optional, but is needed in order for the client to receive
+        /// observer messages from different clusters.
+        /// </summary>
+        public string ClusterId { get; set; }
```

Comment:
On https://github.com/dotnet/orleans/blob/master/src/Orleans/Runtime/OutsideRuntimeClient.cs#L290 client, as part of its initialization in `OutsideRuntimeClient.StartInternal()`, calls to a gateway silo to retrieve the type code map. I think that could be a good place/call to also receive cluster ID from the cluster that client is connecting to, and avoid passing it through client config, which is inherently more error prone.